main/src/modules/tools/opcPopulater/src/opcPopulater.cpp

```diff
@@ -310,6 +376,47 @@ bool opcPopulater::setSaliencyEntity(Bottle bInput){
     return true;
 }
 
+bool opcPopulater::setValueEntity(Bottle bInput){
+
+    if (bInput.size() != 3)
+    {
+        yWarning() << " in opcPopulater::setValueEntity| wrong number of input";
+        return false;
+    }
+
+
+    string sName = bInput.get(1).toString();
+    double targetValue = bInput.get(2).asDouble();
+
+    iCub->opc->checkout();
+    list<Entity*> lEntities = iCub->opc->EntitiesCache();

```

Comment:
The code below is way too complicated! Just do
```
Entity *e = getEntity(sName);
if(e && (e->entity_type() == "agent" || e->entity_type == "object" || e->entity_type == "rtobject")) {
    Object* temp = dynamic_cast<Object*>(e);
    temp->m_value = targetValue;
    iCub->opc->commit();
}
```

It does exactly the same ...